sql/src/main/java/io/crate/operation/collect/StatsTables.java

```diff
@@ -186,58 +177,26 @@ public void operationFinished(@Nullable Integer operationId, @Nullable UUID jobI
     }
 
 
-    public Supplier<Iterable<?>> jobsGetter() {
-        return jobsIterableGetter;
+    public Iterable<?> jobsGetter() {
```

Comment:
while you're at it can you also change this to `Iterable<ConcreteType>`  instead of `Iterable<?>` ?  This will allow us to at some point get some type safety into the  Iterable->Input/Expression machinery